blenderproc/python/writer/BopWriterUtility.py

```diff
@@ -131,14 +133,46 @@ def write_bop(output_dir: str, target_objects: Optional[List[MeshObject]] = None
         chunk_dir_ids = [d.split('/')[-1] for d in chunk_dirs]
         chunk_dirs = chunk_dirs[chunk_dir_ids.index(f"{starting_chunk_id:06d}"):]
 
+        # convert all objects to trimesh objects
+        trimesh_objects = {}
+        for obj in dataset_objects:
+            if isinstance(obj, Link):
+                if not obj.visuals:
+                    continue
+                if len(obj.visuals) > 1:
+                    warnings.warn('BOP Writer only supports saving annotations of one visual mesh per Link')
+            trimesh_obj = obj.mesh_as_trimesh()
+            # we need to create a double-sided material to be able to render non-watertight meshes
+            # the other parameters are defaults, see
+            # https://github.com/mmatl/pyrender/blob/master/pyrender/mesh.py#L216-L223
+            material = pyrender.MetallicRoughnessMaterial(alphaMode='BLEND', baseColorFactor=[0.3, 0.3, 0.3, 1.0],
+                                                          metallicFactor=0.2, roughnessFactor=0.8, doubleSided=True)
+            trimesh_objects[obj.get_cp('category_id')] = pyrender.Mesh.from_trimesh(mesh=trimesh_obj, material=material)
```

Comment:
This loop will run unnecessarily often if you have many objects of the same `category_id` in the scene. So would run a check in line 139:

```python
if obj.get_cp('category_id') in trimesh_objects:
   continue
```

Reply:
yes, but that would not allow a user with duplicate objects in the scene to get bop annotations from them. is that desired?

Reply:
I think this would still work as only one mesh is rendered per scene anyways? Currently you just overwrite the trimesh_objects dict entry in every iteration if there is a duplicate object.

Reply:
oh yes there you have a point, forgot about that. sure, i'll adapt the loop accordingly!